core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/CoarseMesosSchedulerBackend.scala

```diff
@@ -308,6 +303,23 @@ private[spark] class CoarseMesosSchedulerBackend(
     }
   }
 
+    // ToDo: Abstract out each condition and log them.
+  def isOfferSatisfiesRequirements(slaveId: String, mem: Double, cpusOffered: Int,
```

Comment:
The name sounds a bit funny to me. Maybe just `offerSatisfiesRequirements`?

Also, this might be `private`, or `private[spark]` to allow testing.

Ideally the `TODO` would be implemented as well.